bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/HBaseSnapshotInputConfigBuilder.java

```
@@ -0,0 +1,105 @@
+    this.projectId = projectId;
+    return this;
+  }
+
+  public HBaseSnapshotInputConfigBuilder setExportedSnapshotDir(String exportedSnapshotDir) {
+    this.exportedSnapshotDir = exportedSnapshotDir;
+    return this;
+  }
+
+  public HBaseSnapshotInputConfigBuilder setSnapshotName(String snapshotName) {
+    this.snapshotName = snapshotName;
+    return this;
+  }
+
+  public HBaseSnapshotInputConfigBuilder setRestoreDir(String restoreDir) {
+    this.restoreDir = restoreDir;
+    return this;
+  }
+
+  public Configuration build() {
+    Preconditions.checkNotNull(projectId);
+    Preconditions.checkNotNull(exportedSnapshotDir);
+    Preconditions.checkNotNull(snapshotName);
+    Preconditions.checkArgument(
+        exportedSnapshotDir.startsWith("gs://"), "snapshot folder must be hosted in a GCS bucket ");
+
+    Configuration conf = HBaseConfiguration.create();
+    try {
+      conf.set("hbase.rootdir", exportedSnapshotDir);
+      conf.set("fs.AbstractFileSystem.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS");
+      conf.set("fs.gs.project.id", projectId);
+      conf.set("fs.defaultFS", exportedSnapshotDir);
+      conf.set("google.cloud.auth.service.account.enable", "true");
+      conf.setClass(
+          "mapreduce.job.inputformat.class", TableSnapshotInputFormat.class, InputFormat.class);
+      conf.setClass("key.class", ImmutableBytesWritable.class, Writable.class);
+      conf.setClass("value.class", Result.class, Object.class);
+      ClientProtos.Scan proto = ProtobufUtil.toScan(new Scan().setBatch(BATCH_SIZE));
+      conf.set(TableInputFormat.SCAN, Base64.encodeBytes(proto.toByteArray()));
+
+      // LOG.debug(conf);
+      Job job = Job.getInstance(conf); // creates internal clone of hbaseConf
+      TableSnapshotInputFormat.setInput(job, snapshotName, new Path(restoreDir));
+      return job.getConfiguration(); // extract the modified clone
+    } catch (Exception e) {
+      LOG.fatal(e);
```

Why are you swallowing this exception? how can the job have any chance of continuing if this failed?

---

Will throw this and exit the pipeline building